tools/cli/src/commands/Command.ts

```diff
@@ -0,0 +1,166 @@
+import { Err, Ok, Result } from '@backtrace/sourcemap-tools';
+import commandLineArgs from 'command-line-args';
+import commandLineUsage, { Section } from 'command-line-usage';
+import { LoggerOptions, createLogger } from '../logger';
+import { CommandError } from '../models/CommandError';
+import { ExtendedOptionDefinition } from '../models/OptionDefinition';
+
+export class Command<T extends object = any> {
```

Comment:
Does it need to be any? We know what type we expect here. Does it need to be generic?